grails-app/services/org/pih/warehouse/dashboard/IndicatorDataService.groovy

```diff
@@ -1168,6 +1169,83 @@ class IndicatorDataService {
         return getBackdatedShipmentsChart(locationId, monthsLimit, StockMovementDirection.INBOUND)
     }
 
+    @Cacheable(value = "dashboardCache", key = { "getItemsWithBackdatedShipments-${location?.id}" })
+    GraphData getItemsWithBackdatedShipments(Location location) {
+        Integer monthsLimit = Holders.config.openboxes.dashboard.backdatedShipments.monthsLimit
+        Date timeLimit = LocalDate.now().minusMonths(monthsLimit).toDate()
+        Map<String, Integer> amountOfItemsWithBackdatedShipments = [
+            (Constants.ONE)            : 0,
+            (Constants.TWO)            : 0,
+            (Constants.THREE)          : 0,
+            (Constants.FOUR_OR_MORE)   : 0,
+        ]
+
+        String query = '''
+            SELECT ii.product_id, COUNT(ii.product_id) as products, GROUP_CONCAT(bt.shipment_number SEPARATOR ' ') as shipments, DATE_FORMAT(sc.last_stock_count, "%d-%M-%Y") FROM (
+                SELECT t.id as transaction_id, s.shipment_number FROM shipment s
+                INNER JOIN `transaction` t ON t.outgoing_shipment_id  = s.id
+                WHERE DATE_ADD(t.transaction_date, INTERVAL :daysOffset DAY) < t.date_created
+                AND t.date_created > :timeLimit
+                AND s.origin_id = :locationId
+            ) AS bt
+            INNER JOIN transaction_entry te ON te.transaction_id = bt.transaction_id
+            INNER JOIN inventory_item ii ON te.inventory_item_id = ii.id
+            LEFT JOIN (
+                SELECT ii.product_id, MAX(t.transaction_date) as last_stock_count from transaction_entry te 
+                LEFT JOIN inventory_item ii ON te.inventory_item_id = ii.id 
+                LEFT JOIN `transaction` t ON t.id = te.transaction_id
+                LEFT JOIN transaction_type tt ON tt.id = t.transaction_type_id 
+                WHERE t.inventory_id = :inventoryId
+                AND tt.transaction_code = :transactionCode
+                GROUP BY ii.product_id 
+            ) as sc ON sc.product_id = ii.product_id
+            GROUP BY ii.product_id, sc.last_stock_count
+        '''
+
+        List<GroovyRowResult> results = dataService.executeQuery(query, [
+                locationId: location?.id,
+                inventoryId: location?.inventory?.id,
+                transactionCode: TransactionCode.PRODUCT_INVENTORY.name(),
```

Comment:
I think this also should/could include `TransactionCode.INVENTORY`. I think it was used in the past for marking the record stock done through import inventory. At least we are still looking into this type while calculating transactions for product availability and the latest inventory dates indicator.

Reply:
I am not sure about that, this subquery for getting the last stock count date is the same as in the product view (I copied this part of the query from there).
So, this method is called in the view:
![image](https://github.com/openboxes/openboxes/assets/83239466/cde11e33-ae40-460c-80bd-b433fded2df2)
This is the method that contains the query:
![image](https://github.com/openboxes/openboxes/assets/83239466/c72df17f-063d-4298-8225-cc7e020ba697)

But, if you are sure about it, I can change that

Reply:
Sorry just seeing this now. I would prefer NOT to include Inventory. We have removed all of the features that created this type of transaction so if there's data in the database it's very old and somewhat irrelevant. In other words, if a transaction exists that has an "INVENTORY" transaction code I'd be fine for the last stock count feature to return Never.

Reply:
@jmiranda do we have all these types of decisions documented somewhere? I based my comment (suggestion) on the past behavior, and state of the current code where we are still looking into this transaction type, and there is some dead code in the `InventoryService` that is using it and it was a bit misleading. Perhaps we could at least add some comments about it in the `TransactionCode` enum?

Reply:
> and state of the current code where we are still looking into this transaction type

I think you're misread what I said. 

> We have removed all of the features that created this type of transaction

If that's not true and there's still code that **creates** INVENTORY transactions then please flag them and let me know.

> do we have all these types of decisions documented somewhere?

Yes, this particular issue was documented before you guys were involved.
https://github.com/openboxes/openboxes/issues/165

The underlying issue is the QoH Calculation issue that is near and dear to my heart. It was recently documented here but there are probably countless discussions on this issue throughout tickets, documents, and community posts.
https://pihemr.atlassian.net/browse/OBPIH-6353

The pull request for #165 is this epic roller coaster. 
https://github.com/openboxes/openboxes/pull/710

I was only able to get to a place where we could convert the INVENTORY transactions to ADJUSTMENT transactions on a product by product basis. The idea was to eventually move this to an automated feature that would convert everything, but I wasn't comfortable with that since there were some edge cases that I wasn't absolutely sure about. 

So I pushed that code and converted all of the transactions in the PIH and SBHF databases. But I have been holding out hope that I'd someday get become independently wealth so I could fund my own development priorities or have time to work on it as tech debt using PIH's budget. However, it's pretty far down on the tech debt priority list, so I think it'll probably be refactored out completely once we have time to work on the Cycle Count feature and the data model refactorings around that. 

> Perhaps we could at least add some comments about it in the TransactionCode enum?

Yes, that's probably a good idea.